File: Aplikacja_Mobilna/App_IK/app/src/main/java/inteligenty_zamek/app_ik/beforeChange/History_using_keyActivity.java
package inteligenty_zamek.app_ik.beforeChange;

import android.content.res.TypedArray;
import android.os.Bundle;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import org.json.JSONArray;

import java.util.LinkedHashMap;
import java.util.List;

import inteligenty_zamek.app_ik.Navigation.BaseActivity;
import inteligenty_zamek.app_ik.R;


public class History_using_keyActivity extends BaseActivity {

    private String[] navMenuTitles;
    private TypedArray navMenuIcons;
    int toastDelay = 4000;
    JSONArray arrJson;
    LinkedHashMap<String, String> Keys;
    ListView resultsListView;
    LinkedHashMap<String, String> resultsMap;
    List<LinkedHashMap<String, String>> listItems;
    List<LinkedHashMap<String, String>> listItems2;
    boolean flag=true;
    CharSequence csk;
    SimpleAdapter adapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_history_of_using_key);
        navMenuTitles = getResources().getStringArray(R.array.nav_drawer_items);
/*
        navMenuIcons = getResources().obtainTypedArray(R.array.nav_drawer_icons);
        set(navMenuTitles, navMenuIcons);
        resultsListView = (ListView) History_using_keyActivity.this.findViewById(R.id.ListView_History);
        Typeface fontFamily = Typeface.createFromAsset(this.getAssets(), "fonts/fontawesome.ttf");
        TextView sampleText = (TextView) this.findViewById(R.id.TextView_sortingIco);
        sampleText.setTypeface(fontFamily);

        try {
            if (((GlobalClassContainer) getApplication()).getIsadmin() >= 0) {
                User user = ((GlobalClassContainer) getApplication()).getUser();
                new History_using_keyActivity.HTTPRequest(user).execute();
            }
        } catch (Exception except) {}

       EditText inputSearch = (EditText) findViewById(R.id.editText_Search);
        inputSearch.addTextChangedListener(new TextWatcher() {

            @Override
            public void onTextChanged(CharSequence cs, int arg1, int arg2, int arg3) {
                csk=cs;
                Keys = new LinkedHashMap<>();
                listItems2 = new ArrayList<>();
                adapter = new SimpleAdapter(History_using_keyActivity.this, listItems2, R.layout.history_using_key_list,
                        new String[]{"First Line", "Second Line", "Third Line"},
                        new int[]{R.id.history_icon, R.id.title, R.id.describe});
                try {
                    for (int i = 0; i < listItems.size(); i++) {
                        Keys.put(listItems.get(i).values().toArray()[1].toString(), listItems.get(i).values().toArray()[2].toString());
                        Log.i(listItems.get(i).values().toArray()[1].toString(),listItems.get(i).values().toArray()[2].toString());

                    }
                }catch (Exception e) {}
                Iterator it = Keys.entrySet().iterator();
                while (it.hasNext())
                {
                    resultsMap = new LinkedHashMap<>();
                    Map.Entry pair = (Map.Entry)it.next();
                    Log.i("aaaaaaa",pair.getValue().toString().toLowerCase());
                    Log.i("aaaaaaaa",cs.toString().toLowerCase());
                    if(
                            pair.getKey().toString().toLowerCase().contains(cs.toString().toLowerCase())
                                    || pair.getValue().toString().toLowerCase().contains(cs.toString().toLowerCase())
                            ) {


                        resultsMap.put("First Line", "");
                        resultsMap.put("Second Line", pair.getKey().toString());
                        resultsMap.put("Third Line", pair.getValue().toString());
                        listItems2.add(resultsMap);
                    }
                    else if(cs.toString()=="")
                    {


                        resultsMap.put("First Line", "");
                        resultsMap.put("Second Line", pair.getKey().toString());
                        resultsMap.put("Third Line", pair.getValue().toString());
                        listItems2.add(resultsMap);
                    }
                }
                resultsListView.setAdapter(adapter);
            }

            @Override
            public void beforeTextChanged(CharSequence arg0, int arg1, int arg2,
                                          int arg3) {
                // TODO Auto-generated method stub

            }

            @Override
            public void afterTextChanged(Editable arg0) {
                // TODO Auto-generated method stub
            }
        });


        //przycisk dopowiedzialny za sortowanie
        TextView  textView= (TextView) findViewById(R.id.TextView_sortingIco);
        textView.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View viewIn) {




                if(flag==false) {
                    Arrays.sort(listItems2.toArray());
                    flag=true;
                }
                else
                {
                    //Comparator <LinkedHashMap<String, String>> cmp =
                            Collections.sort(listItems2, new Comparator<LinkedHashMap<String, String>>(){
                        public int compare(LinkedHashMap<String, String> one, LinkedHashMap<String, String> two) {
                            return one.values().toString().compareTo(two.values().toString());
                            }});



                 Arrays.sort(listItems2.toArray());
                    flag=false;
                }

                Keys =new LinkedHashMap<>();
                 adapter = new SimpleAdapter(History_using_keyActivity.this, listItems2, R.layout.history_using_key_list,
                        new String[]{"First Line", "Second Line", "Third Line"},
                        new int[]{R.id.history_icon, R.id.title, R.id.describe});
               // try {
                 //   for (int i = 0; i < ((GlobalClassContainer) getApplication()).getUser().getCertyficateList(this).length; i++) {
                  //      Keys.put(listItems2.get(i).values().toArray()[0].toString(), listItems2.get(i).values().toArray()[1].toString() );
                    }
               // }catch (Exception e) {}
                Iterator it = Keys.entrySet().iterator();

                while (it.hasNext())
                {
                    Log.i("aaaaaaa","tutaj3");
                    resultsMap = new LinkedHashMap<>();
                    Map.Entry pair = (Map.Entry)it.next();
                    if(
                            pair.getKey().toString().toLowerCase().contains(csk.toString().toLowerCase())
                                    || pair.getValue().toString().toLowerCase().contains(csk.toString().toLowerCase())
                            ) {
                        Log.i("bbbb", pair.getKey().toString());
                        resultsMap.put("First Line", "");
                        resultsMap.put("Second Line", pair.getKey().toString());
                        resultsMap.put("Third Line", pair.getValue().toString());
                        listItems.add(resultsMap);
                    }
                }
                resultsListView.setAdapter(adapter);
            }
        });















    }


    public class HTTPRequest extends AsyncTask<Void, Void, String> {
        User user;
        boolean choise;

        public HTTPRequest(User x) {
            user = x;
        }

        @Override
        protected String doInBackground(Void... params) {
            HttpClient httpclient = new DefaultHttpClient();
            String adres = "http://" + ((GlobalClassContainer) getApplication()).getSerwerIP() + ":8080/api/admin/history/";
            HttpPost httppost = new HttpPost(adres);
            try {
                // Add your data
                List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>(2);
                nameValuePairs.add(new BasicNameValuePair("login", user.getLogin()));
                nameValuePairs.add(new BasicNameValuePair("token", ((GlobalClassContainer) getApplication()).getSession()));
                httppost.setEntity(new UrlEncodedFormEntity(nameValuePairs));
                // Execute HTTP Post Request
                HttpResponse response = httpclient.execute(httppost);
                HttpEntity entity = response.getEntity();
                String responseString = EntityUtils.toString(entity, "UTF-8");
                return responseString;
            } catch (ClientProtocolException e) {
                // TODO Auto-generated catch block
            } catch (IOException e) {
                // TODO Auto-generated catch block
            }
            return null;
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        @Override
        protected void onPostExecute(String response) {
            super.onPostExecute(response);
            JSONObject jObj = null;
            try {
                jObj = new JSONObject(response);
                arrJson = jObj.getJSONArray("data");
                 //hasmap przechowujący elemety do wyświetlenia
                 Keys = new LinkedHashMap<>();
                for (int i = 0; i < arrJson.length(); i++) {
                    try {
                        String[] date = arrJson.getJSONObject(i).getString("DATE").split("T");
                        if (arrJson.getJSONObject(i).getInt("ACCESS") == 1) {
                            Keys.put("Otwarcie " + Integer.toString(i), "zamek " + arrJson.getJSONObject(i).getString("ZAMEK") + " został otwarty przez " + arrJson.getJSONObject(i).getString("NAME") + " " + arrJson.getJSONObject(i).getString("SURNAME") + ", data zdarzenia: " + date[0] +  " godzina: " + date[1]);
                        } else {
                            Keys.put("Nieautoryzowana próba otwarcia " + Integer.toString(i), "zamek " + arrJson.getJSONObject(i).getString("ZAMEK") + " zarejestrował nieutoryzowaną próbę dostępu przez " + arrJson.getJSONObject(i).getString("NAME") + " " + arrJson.getJSONObject(i).getString("SURNAME") + ", data zdarzenia: " + date[0] +  " godzina: " + date[1]);
                        }
                    } catch (JSONException ignored) {}
                }
                //stworzenie adaptera
                listItems = new ArrayList<>();
                adapter = new SimpleAdapter(History_using_keyActivity.this, listItems, R.layout.history_using_key_list,
                        new String[]{"First Line", "Second Line", "Third Line"},
                        new int[]{R.id.history_icon, R.id.title, R.id.describe});


                //TODO zmiana koloru czcionki nieautoryzowany dostęp
                String buffor;
                //iterator elementow (przepisanie z hashmap do adaptera[listitems] elementow)
                Iterator it = Keys.entrySet().iterator();
                while (it.hasNext()) {
                    resultsMap = new LinkedHashMap<>();
                    Map.Entry pair = (Map.Entry) it.next();
                    buffor = pair.getKey().toString();
                    if (buffor == "Nieautoryzowana próba otwarcia") {
                        resultsMap.put("First Line", "");
                    } else {
                        resultsMap.put("First Line", "");
                    }
                    resultsMap.put("Second Line", buffor);
                    resultsMap.put("Third Line", pair.getValue().toString());
                    listItems.add(resultsMap);
                }
                listItems2=listItems;
                resultsListView.setAdapter(adapter);
                final Toast toast = Toast.makeText(History_using_keyActivity.this, "Pobrano historie zamków", Toast.LENGTH_LONG);
                toast.show();
                new CountDownTimer(toastDelay, 1000) {
                    public void onTick(long millisUntilFinished) {
                        toast.show();
                    }

                    public void onFinish() {
                        toast.show();
                    }
                }.start();

            } catch (JSONException e) {

                final Toast toast = Toast.makeText(History_using_keyActivity.this, "Niemożna pobrać historii zamków", Toast.LENGTH_LONG);
                toast.show();
                new CountDownTimer(toastDelay, 1000) {
                    public void onTick(long millisUntilFinished) {
                        toast.show();
                    }

                    public void onFinish() {
                        toast.show();
                    }
                }.start();
            }
        }
    }
}*/
    }
    }

File: Aplikacja_Mobilna/App_IK/app/src/main/java/inteligenty_zamek/app_ik/inWork/History_using_keyActivity.kt
package inteligenty_zamek.app_ik.inWork

import android.content.Context
import android.content.res.TypedArray
import android.graphics.Typeface
import android.os.AsyncTask
import android.os.Bundle
import android.os.CountDownTimer
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.text.Editable
import android.text.TextWatcher
import android.util.Log
import android.view.View
import android.widget.*

import org.apache.http.NameValuePair
import org.apache.http.client.ClientProtocolException
import org.apache.http.client.entity.UrlEncodedFormEntity
import org.apache.http.client.methods.HttpPost
import org.apache.http.impl.client.DefaultHttpClient
import org.apache.http.message.BasicNameValuePair
import org.apache.http.util.EntityUtils
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.io.IOException
import java.util.ArrayList
import java.util.Arrays
import java.util.Collections
import java.util.LinkedHashMap

import inteligenty_zamek.app_ik.Navigation.BaseActivity
import inteligenty_zamek.app_ik.R
import inteligenty_zamek.app_ik.adapters.HistoryListAdapter
import inteligenty_zamek.app_ik.adapters.TwoButtonAdapter
import inteligenty_zamek.app_ik.adapters.userWatModel
import inteligenty_zamek.app_ik.models.HistoryListElement
import inteligenty_zamek.app_ik.rest_class.GlobalClassContainer
import inteligenty_zamek.app_ik.rest_class.User
import java.security.AccessController.getContext
import java.util.Map


class History_using_keyActivity : BaseActivity() {

    var navMenuTitles: Array<String>? = null
    var navMenuIcons: TypedArray? = null
    var toastDelay = 4000
    var arrJson: JSONArray? = null
    var Keys: LinkedHashMap<String, String>?=null
    var resultsListView: RecyclerView?=null
    var resultsMap: LinkedHashMap<String, String>?=null
    var listItems: MutableList<LinkedHashMap<String, String>>? = null
    var listItems2: MutableList<LinkedHashMap<String, String>>?=null
    var flag = true
    var csk: CharSequence=""
    var adapter: HistoryListAdapter?=null
    var presenter:History_using_keyPresenter?=null
    var firstUse=false
    var context:Context?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_history_of_using_key)
        navMenuTitles = resources.getStringArray(R.array.nav_drawer_items)

        navMenuIcons = resources.obtainTypedArray(R.array.nav_drawer_icons)
        set(navMenuTitles, navMenuIcons)
        resultsListView = this@History_using_keyActivity.findViewById(R.id.historyUsingKeyRecyclerView) as RecyclerView
        val fontFamily = Typeface.createFromAsset(this.assets, "fonts/fontawesome.ttf")
        val sampleText = this.findViewById(R.id.TextView_sortingIco) as TextView
        sampleText.typeface = fontFamily
        context=this
        presenter=History_using_keyPresenter(this)

        presenter!!.getList()
        (findViewById(R.id.historyUsingKeyTextViewShowHideIco) as TextView).typeface=fontFamily

    var textView = findViewById(R.id.historyUsingKeyLinearLayoutSorting) as LinearLayout
    textView.setOnClickListener( View.OnClickListener {

        val scale = this.getResources().getDisplayMetrics().density
        val size = (30 * scale + 0.5f)

        if(textView.getLayoutParams().height==size.toInt()) {
            val params = textView.getLayoutParams()
            params.height = LinearLayout.LayoutParams.WRAP_CONTENT;
            textView.setLayoutParams(params)
            (findViewById(R.id.historyUsingKeyTextViewShowHideIco) as TextView).setText("\uf106")
            (findViewById(R.id.historyUsingKeyTextViewShowHideText) as TextView).setText("zwiń wyszukiwanie")

        }
        else
        {
            val params = textView.getLayoutParams()
            params.height = size.toInt()
            textView.setLayoutParams(params)
            (findViewById(R.id.historyUsingKeyTextViewShowHideIco) as TextView).setText("\uf107")
            (findViewById(R.id.historyUsingKeyTextViewShowHideText) as TextView).setText("rozwiń wyszukiwanie")
        }
    })


//

/*
val inputSearch = findViewById(R.id.HistoryEditText) as EditText
inputSearch.addTextChangedListener(object : TextWatcher {

   override fun onTextChanged(cs: CharSequence, arg1: Int, arg2: Int, arg3: Int) {
       csk = cs
       Keys = LinkedHashMap()
       listItems2 = ArrayList()
       adapter = SimpleAdapter(this@History_using_keyActivity, listItems2, R.layout.history_using_key_list,
               arrayOf("First Line", "Second Line", "Third Line"),
               intArrayOf(R.id.history_icon, R.id.title, R.id.describe))
       try {
           for (i in listItems!!.indices) {
              presenter!!.model. Keys!!.put(listItems!![i].values.toTypedArray()[1].toString(), listItems!![i].values.toTypedArray()[2].toString())
               Log.i(listItems!![i].values.toTypedArray()[1].toString(), listItems!![i].values.toTypedArray()[2].toString())

           }
       } catch (e: Exception) {
       }

       val it = Keys!!.entries.iterator()
       while (it.hasNext()) {
           resultsMap = LinkedHashMap()
           val pair = it.next() as Map.Entry<*, *>
           Log.i("aaaaaaa", pair.value.toString().toLowerCase())
           Log.i("aaaaaaaa", cs.toString().toLowerCase())
           if (pair.key.toString().toLowerCase().contains(cs.toString().toLowerCase()) || pair.value.toString().toLowerCase().contains(cs.toString().toLowerCase())) {


               resultsMap!!.put("First Line", "")
               resultsMap!!.put("Second Line", pair.key.toString())
               resultsMap!!.put("Third Line", pair.value.toString())
               listItems2!!.add(resultsMap!!)
           } else if (cs.toString() === "") {


               resultsMap!!.put("First Line", "")
               resultsMap!!.put("Second Line", pair.key.toString())
               resultsMap!!.put("Third Line", pair.value.toString())
               listItems2!!.add(resultsMap!!)
           }
       }
       resultsListView!!.adapter = adapter
   }

   override fun beforeTextChanged(arg0: CharSequence, arg1: Int, arg2: Int,
                                  arg3: Int) {
       // TODO Auto-generated method stub

   }

   override fun afterTextChanged(arg0: Editable) {
       // TODO Auto-generated method stub
   }
})
*/

//przycisk dopowiedzialny za sortowanie
/*val textView = findViewById(R.id.TextView_sortingIco) as TextView
textView.setOnClickListener(object : View.OnClickListener {
   // }catch (Exception e) {}
   internal var it: Iterator<*> = Keys!!.entries.iterator()

   override fun onClick(viewIn: View) {


       if (flag == false) {
           Arrays.sort(listItems2!!.toTypedArray())
           flag = true
       } else {
           //Comparator <LinkedHashMap<String, String>> cmp =
           Collections.sort(listItems2) { one, two -> one.values.toString().compareTo(two.values.toString()) }



           Arrays.sort(listItems2!!.toTypedArray())
           flag = false
       }

       Keys = LinkedHashMap()
       adapter = SimpleAdapter(this@History_using_keyActivity, listItems2, R.layout.history_using_key_list,
               arrayOf("First Line", "Second Line", "Third Line"),
               intArrayOf(R.id.history_icon, R.id.title, R.id.describe))
       // try {
       //   for (int i = 0; i < ((GlobalClassContainer) getApplication()).getUser().getCertyficateList(this).length; i++) {
       //      Keys.put(listItems2.get(i).values().toArray()[0].toString(), listItems2.get(i).values().toArray()[1].toString() );

   }

   init {
       resultsMap = LinkedHashMap()
       val pair = it.next() as Map.Entry<*, *>
       if (pair.key.toString().toLowerCase().contains(csk.toString().toLowerCase()) || pair.value.toString().toLowerCase().contains(csk.toString().toLowerCase())) {
           Log.i("bbbb", pair.key.toString())
           resultsMap!!.put("First Line", "")
           resultsMap!!.put("Second Line", pair.key.toString())
           resultsMap!!.put("Third Line", pair.value.toString())
           listItems!!.add(resultsMap!!)
       }
   }
})*/

}
fun setAdapter(listitem:ArrayList<HistoryListElement>)
{
Log.i("HHHH","bylem w setAdapter")
val recyclerView = (findViewById(R.id.historyUsingKeyRecyclerView) as RecyclerView)
adapter = HistoryListAdapter(listitem,context)
val mLayoutManager = LinearLayoutManager(applicationContext)
recyclerView!!.setLayoutManager(mLayoutManager)
recyclerView!!.setItemAnimator(DefaultItemAnimator())
recyclerView!!.setAdapter(adapter)
if(!firstUse)
{firstUse=true
recyclerView!!.addItemDecoration(DividerItemDecoration(this, LinearLayoutManager.VERTICAL))}
adapter!!.notifyDataSetChanged()

// adapter = SimpleAdapter(this@History_using_keyActivity,
//       presenter!!.model!!.listItems, R.layout.history_using_key_list,
//         arrayOf("First Line", "Second Line", "Third Line"),
//         intArrayOf(R.id.history_icon, R.id.title, R.id.describe))
// resultsListView!!.setAdapter(adapter)
}

}




